File: android/services/kithara/src/main/java/com/mplayeraudio/services/kithara/AudioEngineModels.kt
package com.mplayeraudio.services.kithara

data class AudioTrackRequest(
    val id: String,
    val url: String,
)

data class AudioEngineState(
    val status: AudioEngineStatus = AudioEngineStatus.Idle,
    val currentPositionMs: Long = 0L,
    val durationMs: Long? = null,
    val bufferedPositionMs: Long = 0L,
    val currentItemId: String? = null,
    val isPlaying: Boolean = false,
    val error: AudioEngineError? = null,
)

enum class AudioEngineStatus {
    Idle,
    ReadyToPlay,
    Failed,
}

sealed interface AudioEngineEvent {
    data class CurrentItemChanged(val itemId: String?) : AudioEngineEvent
    data object PlayedToEnd : AudioEngineEvent
    data class ItemFailed(val itemId: String, val reason: AudioEngineError) : AudioEngineEvent
    data class EngineFailed(val reason: AudioEngineError) : AudioEngineEvent
    data class DurationDiscovered(val itemId: String, val durationMs: Long) : AudioEngineEvent
}

sealed interface AudioEngineError {
    data class LoadFailed(val message: String) : AudioEngineError
    data class StreamFailed(val message: String) : AudioEngineError
    data class EngineCrashed(val message: String) : AudioEngineError
    data object SeekFailed : AudioEngineError
}

File: ios/Services/KitharaPlayer/Sources/ServicesKitharaPlayer/AudioEngineModels.swift
import Foundation

struct AudioTrackRequest: Sendable {
    let id: String
    let url: String
}

struct AudioEngineState: Sendable {
    var status: AudioEngineStatus = .idle
    var currentPositionMs: Int64 = 0
    var durationMs: Int64?
    var currentItemId: String?
    var isPlaying = false
    var errorMessage: String?
}

enum AudioEngineStatus: Sendable {
    case idle
    case readyToPlay
    case failed
}

enum AudioEngineEvent: Sendable {
    case currentItemChanged(itemId: String?)
    case playedToEnd
    case itemFailed(itemId: String?, reason: String)
}
